Solution1/authentication_app/Program.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+
+// Пользователю предлагается ввести имя пользователя и пароль, при совпадении имени пользователя
+// и пароля с имеющимися в системе – выдать сообщение об успешном логине, при несовпадении –
+// выдать сообщение об ошибке и предложить еще одну попытку. 
+// Максимальное количество попыток – 3, при 3 неуспешных попытках авторизации – выдать сообщение об
+// ошибке, закрыть приложение после нажатия любого символа.
+
+namespace AuthenticationApp
+{
+    class Program
+    {
+        static bool CheckAuthentication()
+        {
+            string user = "admin";
+            string password = "pass";
+
+            Console.WriteLine("Enter user name:");
+            string userName = Console.ReadLine();
+
+            Console.WriteLine("Enter password:");
+            string userPass = Console.ReadLine();
+
+            bool credMatch = (userName == user) && (userPass == password);
+
+            return credMatch;
+        }
+
+        static void Main()
+        {
+            int i = 0;
+            do
+            {
+                if (CheckAuthentication() == true)
+                {
+                    Console.WriteLine("Authentication successful!");
+                    break;
+                }
+                else if (i < 2)
```

Comment:
Should't be else if
You've got 2 different checks here, first about the successful or wrong authentication and second about state of the iterator, so just 2 ifs

Reply:
Done